src/main/java/decodes/dbeditor/routing/RSListTableModel.java

```diff
@@ -112,6 +112,16 @@ public void addObject(RoutingSpec ob)
         fireTableDataChanged();
     }
 
+    public void addOrReplace(RoutingSpec ob)
+    {
+        RoutingSpec found = theList.find(ob.getName());
+        if (found != null)
+        {
+            theList.remove(found);
```

Comment:
does the database version also need to be deleted here? (call deleteObject(RoutingSpec ob)

Reply:
This is specifically *not* calling the database delete, as "add" doesn't call the database write, and thus the Spec will magically disappear from the database.

There's a lot of things to fix, or just make a firm decision about, in the table models.